grafana/README.md

```diff
@@ -1,87 +1,43 @@
-INFLUXDB_HOST=**ChangeMe**          Host of your InfluxDB (without protocol)
-INFLUXDB_PORT=8086                  Port number of your InfluxDB
-INFLUXDB_NAME=**ChangeMe**          Database name of your InfluxDB
-INFLUXDB_USER=root                  Username of your InfluxDB
-INFLUXDB_PASS=root                  Password of your InfluxDB
-```
-
-Here is an example:
-
-    docker run -d -p 80:80 -e INFLUXDB_HOST=influxdb-1-tifayuki.delta.tutum.io -e INFLUXDB_PORT=8086 -e INFLUXDB_NAME=testdb -e INFLUXDB_USER=root -e INFLUXDB_PASS=root tutum/grafana
-
-
-Configure Elasticsearch to save and load dashboards
----------------------------------------------------
-If you want you use Elasticsearch to save and load you dashboards, you can provide the following environment variables for configuration:
-
-```
-ELASTICSEARCH_PROTO=http            Protocol of your Elasticsearch
-ELASTICSEARCH_HOST=**None**         Host of your Elasticsearch (without protocol)
-ELASTICSEARCH_PORT=9200             Port number of your Elasticsearch
-ELASTICSEARCH_USER=**None**         Username for elasticsearch if it has HTTP basic auth enabled (leave it to **None** if no HTTP basic auth is needed)
-ELASTICSEARCH_PASS=**None**         Password for elasticsearch if it has HTTP basic auth enabled (leave it to **None** if no HTTP basic auth is needed)
+# Grafana Docker Image
+
+This image runs [Grafana](http://grafana.org) listening on port 3000.
+The image weighs in at very lightweight ~30MB & requires only ~4MB RAM
+to run as it uses a very small golang binary to serve the static files.
+
+The Grafana `config.js` is configured via the following environment
+variables:
+
+* `INFLUXDB_URL` - the URL to the InfluxDB instance you want to connect to
+* `INFLUXDB_SERVICE_NAME` - the name of the Kubernetes InfluxDB service (default: `INFLUXDB`)
+* `INFLUXDB_PROTO` - the protocol to use (default: `http`)
+* `INFLUXDB_NAME` - the name of the InfluxDB database to query (default: `k8s`)
+* `GRAFANA_DB_NAME` - the name of the InfluxDB database to use for Grafana dashboard storage (default: `grafana`)
+* `INFLUXDB_USER` - the username to connect with (default: `root`)
+* `INFLUXDB_PASSWORD` - the password to connect with (default: `root`)
+
+## Running in Kubernetes/OpenShift3
+
+`INFLUXDB_URL` environment variable can itself contain environment variables
+which will be expanded at runtime to populate the `config.js`. This is very
+useful in a [Kubernetes](http://kubernetes.io/) environment where you might
+have an InfluxDB service running. For example, if you have an InfluxDB service
+running called "InfluxDB", then you can use the following configuration to run
+this Grafana image:
+
+```yaml
+containers:
+  - name: grafana-image
+    image: kubernetes/heapster_grafana:latest
+    ports:
+      - name: grafana-port
+        containerPort: 3000
+    env:
+      - name: INFLUXDB_URL
+        value: http://${INFLUX_MASTER_SERVICE_HOST}:${INFLUX_MASTER_SERVICE_PORT}
```

Comment:
Runtime expansion sounds awesome. But DNS should take care of this. So in kubernetes, this could just be the Influxdb service name.